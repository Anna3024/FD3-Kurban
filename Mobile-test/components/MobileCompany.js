import React from 'react';
import PropTypes from 'prop-types';

import MobileClient from './MobileClient';
import {mobileEvents} from './events';

import './MobileCompany.css';

class MobileCompany extends React.PureComponent {

  static propTypes = {
    name: PropTypes.string.isRequired,
    clients:PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        surname: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
        midName: PropTypes.string.isRequired,
        balance: PropTypes.number.isRequired,
      })
    ),
  };

  state = {
    clients: this.props.clients.slice(),//[клиенты ]
    showClientsState: 1, //1 - all, 2 - active, 3 - blocked
    newClientState: false,//поле с новым клентом
    lastId: this.props.clients[this.props.clients.length-1].id 
  };

  componentDidMount = () => {
    mobileEvents.addListener('EDelClient',this.deliteClient); 
    mobileEvents.addListener('ESaveEditedClient',this.saveEditedClient);
  };

  componentWillUnmount = () => {
    mobileEvents.removeListener('EDelClient',this.deliteClient);
    mobileEvents.removeListener('ESaveEditedClient',this.saveEditedClient);
  };

  showClients(num)  {
    this.setState({showClientsState: num})
  }

  getCurrentClientsArr = () => {
    if (this.state.showClientsState==2) {
      return this.state.clients.filter((v)=>v.balance>0)
    }

    if (this.state.showClientsState==3) {
      return this.state.clients.filter((v)=>v.balance<=0)
    }

    return this.state.clients
  }

  deliteClient = (id) => {
    if ( this.state.clients.findIndex(v=>v.id==id)==-1) { //для нового клиента "отменить"
      this.setState({newClientState: false, lastId: this.state.lastId-1});
      return
    }
    let currentClients = [...this.state.clients]; //для удаления клиента из списка
    currentClients = currentClients.filter((v)=>v.id!=id);
    this.setState({clients: currentClients})
  }

  saveEditedClient = (editedClient) => {
    
    if (this.state.clients.findIndex(v=>v.id==editedClient.id)==-1) { //для нового клиента
      this.state.clients.push(editedClient);
      this.setState({newClientState: false});
      return
    }
    let currentClients = [...this.state.clients]; //редактирование клиентов из спписка
    for (let i=0; i< currentClients.length; i++) {
      if ( currentClients[i].id==editedClient.id) {
        currentClients[i]=editedClient;
        this.setState({clients: currentClients})
        break;
      }
    }
  }

  addNewClient = () => {
    this.setState({newClientState: true, lastId: this.state.lastId+1})
  }
  
  render() {

    console.log("MobileCompany render");

    let tableHeader = ['Фамилия', 'Имя', 'Отчество', 'Баланс', 'Статус', 'Редактировать', 'Удалить'];

    var clientsCode=this.getCurrentClientsArr().map( v =>
      <MobileClient key={v.id} client={v} />
    );

    return (
      <div className='MobileCompany'>
        <div className='MobileCompanyName'>Компания: &laquo;{this.props.name}&raquo;</div>
        <div className='MobileControls'>
          <input type="button" className={this.state.showClientsState==1 ? "activeBtn" : null} value="Все" onClick={this.showClients.bind(this,1)} />
          <input type="button" className={this.state.showClientsState==2 ? "activeBtn" : null} value="Активные" onClick={this.showClients.bind(this,2)} />
          <input type="button" className={this.state.showClientsState==3 ? "activeBtn" : null} value="Заблокированные" onClick={this.showClients.bind(this,3)} />
        </div>
        <table>
          <thead>
            <tr>
              {tableHeader.map((v,i)=><th key={i}>{v}</th>)}
            </tr>
          </thead>
          <tbody>
            {clientsCode}
            {this.state.newClientState && //если добавить нового пользователя
            <MobileClient key={this.state.lastId} client={{id: this.state.lastId}}/> //поле для ввода
            }
          </tbody>
        </table>
        <input type="button" value="Добавить клиента" onClick={this.addNewClient} disabled={this.state.newClientState}/>
      </div>
    )
    ;
  }

}

export default MobileCompany;
