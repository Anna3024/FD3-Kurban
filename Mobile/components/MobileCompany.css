.MobileCompany input 
{
  margin: 5px;
}

.MobileCompany .activeBtn 
{
  background-color: #28a745;
}

.MobileCompanyName
{
  font-weight: bold;
  font-size: 20px;
  padding: 5px;
}

.MobileControls, 
.MobileCompanyName,
.MobileCompany tr
{
  margin-left: 5px;
  border-bottom: 1px solid #dee2e6;
}

.MobileCompany table
{
  width: 99%;
  text-align: left;
  margin: 5px;
  border-collapse: collapse;
}

.MobileCompany thead
{
  border-bottom: 2px solid #dee2e6;
}

.MobileCompany th
{
  padding: 5px;
  padding-bottom: 10px;
}



