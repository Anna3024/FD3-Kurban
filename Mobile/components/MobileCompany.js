import React from 'react';
import PropTypes from 'prop-types';

import MobileClient from './MobileClient';
import {mobileEvents} from './events';

import './MobileCompany.css';

class MobileCompany extends React.PureComponent {

  static propTypes = {
    name: PropTypes.string.isRequired,
    clients:PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        surname: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
        midName: PropTypes.string.isRequired,
        balance: PropTypes.number.isRequired,
      })
    ),
  };

  state = {
    name: this.props.name, 
    activeCompany: 0, //0 - velcom, 1 - MTC
    clients: [this.props.clients.slice(), this.props.clients.slice()],//[клиенты velcom, клиенты мтс]
    showClientsState: 1, //1 - all, 2 - active, 3 - blocked
    newClientState: false,//поле с новым клентом
    lastId: this.props.clients[this.props.clients.length-1].id 
  };

  componentDidMount = () => {
    mobileEvents.addListener('EDelClient',this.deliteClient); 
    mobileEvents.addListener('ESaveEditedClient',this.saveEditedClient);
  };

  componentWillUnmount = () => {
    mobileEvents.removeListener('EDelClient',this.deliteClient);
    mobileEvents.removeListener('ESaveEditedClient',this.saveEditedClient);
  };

  setName1 = () => {
    this.setState({name:'МТС', activeCompany: 1});
  };

  setName2 = () => {
    this.setState({name:'Velcom', activeCompany: 0});
  };

  showClients(num)  {
    this.setState({showClientsState: num})
  }

  deliteClient = (id) => {
    if ( this.state.clients[this.state.activeCompany].findIndex(v=>v.id==id)==-1) { //для нового клиента "отменить"
      this.setState({newClientState: false, lastId: this.state.lastId-1});
      return
    }
    let currentClients = [...this.state.clients]; //для удаления клиента из списка
    currentClients[this.state.activeCompany] = currentClients[this.state.activeCompany].filter((v)=>v.id!=id);
    this.setState({clients: currentClients})
  }

  saveEditedClient = (editedClient) => {
    
    if (this.state.clients[this.state.activeCompany].findIndex(v=>v.id==editedClient.id)==-1) { //для нового клиента
      this.state.clients[this.state.activeCompany].push(editedClient);
      this.setState({newClientState: false});
      return
    }
    let currentClients = [...this.state.clients]; //редактирование клиентов из спписка
    for (let i=0; i< currentClients[this.state.activeCompany].length; i++) {
      if ( currentClients[this.state.activeCompany][i].id==editedClient.id) {
        currentClients[this.state.activeCompany][i]=editedClient;
        this.setState({clients: currentClients})
        break;
      }
    }
  }

  addNewClient = () => {
    this.setState({newClientState: true, lastId: this.state.lastId+1})
  }
  
  render() {

    console.log("MobileCompany render");

    let tableHeader = ['Фамилия', 'Имя', 'Отчество', 'Баланс', 'Статус', 'Редактировать', 'Удалить'];

    let currClients = this.state.clients[this.state.activeCompany];

    if (this.state.showClientsState==2) {
      currClients = currClients.filter((v)=>v.balance>0)
    }

    if (this.state.showClientsState==3) {
      currClients = currClients.filter((v)=>v.balance<=0)
    }

    var clientsCode=currClients.map( v =>
      <MobileClient key={v.id} client={v} />
    );

    return (
      <div className='MobileCompany'>
        <input type="button" value="МТС" onClick={this.setName1} />
        <input type="button" value="Velcom" onClick={this.setName2} />
        <div className='MobileCompanyName'>Компания: &laquo;{this.state.name}&raquo;</div>
        <div className='MobileControls'>
          <input type="button" className={this.state.showClientsState==1 ? "activeBtn" : null} value="Все" onClick={this.showClients.bind(this,1)} />
          <input type="button" className={this.state.showClientsState==2 ? "activeBtn" : null} value="Активные" onClick={this.showClients.bind(this,2)} />
          <input type="button" className={this.state.showClientsState==3 ? "activeBtn" : null} value="Заблокированные" onClick={this.showClients.bind(this,3)} />
        </div>
        <table>
          <thead>
            <tr>
              {tableHeader.map((v,i)=><th key={i}>{v}</th>)}
            </tr>
          </thead>
          <tbody>
            {clientsCode}
            {this.state.newClientState && //если добавить нового пользователя
            <MobileClient key={this.state.lastId} client={{id: this.state.lastId}}/> //поле для ввода
            }
          </tbody>
        </table>
        <input type="button" value="Добавить клиента" onClick={this.addNewClient} disabled={this.state.newClientState}/>
      </div>
    )
    ;
  }

}

export default MobileCompany;
