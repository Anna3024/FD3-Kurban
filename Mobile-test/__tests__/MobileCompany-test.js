"use strict";

import React from 'react';
import renderer from 'react-test-renderer';


import clientsArr from '../clientsArr.json';
import MobileCompany from '../components/MobileCompany';

test('работа MobileCompany', ()=>{

    // создаём тестовую версию компонента
    const component = renderer.create(
        <MobileCompany name='Velcom' clients={clientsArr}/>
    );

    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    const btnShowAll = component.root.find( el => el.props.value=="Все");
    btnShowAll.props.onClick();
    // получаем уже изменённый снэпшот
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    const btnShowActive = component.root.find( el => el.props.value=="Активные");
    btnShowActive.props.onClick();
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    const btnShowBlock = component.root.find( el => el.props.value=="Заблокированные");
    btnShowBlock.props.onClick();
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    const btnAdd = component.root.find( el => el.props.value=="Добавить клиента");
    btnAdd.props.onClick();
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

    btnAdd.props.onClick();
    componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
        
})