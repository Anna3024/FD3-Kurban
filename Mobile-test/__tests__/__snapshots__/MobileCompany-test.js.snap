// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа MobileCompany 1`] = `
<div
  className="MobileCompany"
>
  <div
    className="MobileCompanyName"
  >
    Компания: «
    Velcom
    »
  </div>
  <div
    className="MobileControls"
  >
    <input
      className="activeBtn"
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      className={null}
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      className={null}
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
  </div>
  <table>
    <thead>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="MobileClient"
      >
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Иванов
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Иван
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Иванович
        </td>
        <td
          style={
            Object {
              "width": "100px",
            }
          }
        >
          200
        </td>
        <td
          className="active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Сидоров
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Сидор
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Сидорович
        </td>
        <td
          style={
            Object {
              "width": "100px",
            }
          }
        >
          250
        </td>
        <td
          className="active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Петров
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Петр
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Петрович
        </td>
        <td
          style={
            Object {
              "width": "100px",
            }
          }
        >
          180
        </td>
        <td
          className="active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Григорьев
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Григорий
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Григорьевич
        </td>
        <td
          style={
            Object {
              "width": "100px",
            }
          }
        >
          -220
        </td>
        <td
          className="blocked"
        >
          block
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    disabled={false}
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа MobileCompany 2`] = `
<div
  className="MobileCompany"
>
  <div
    className="MobileCompanyName"
  >
    Компания: «
    Velcom
    »
  </div>
  <div
    className="MobileControls"
  >
    <input
      className="activeBtn"
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      className={null}
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      className={null}
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
  </div>
  <table>
    <thead>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="MobileClient"
      >
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Иванов
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Иван
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Иванович
        </td>
        <td
          style={
            Object {
              "width": "100px",
            }
          }
        >
          200
        </td>
        <td
          className="active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Сидоров
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Сидор
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Сидорович
        </td>
        <td
          style={
            Object {
              "width": "100px",
            }
          }
        >
          250
        </td>
        <td
          className="active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Петров
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Петр
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Петрович
        </td>
        <td
          style={
            Object {
              "width": "100px",
            }
          }
        >
          180
        </td>
        <td
          className="active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Григорьев
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Григорий
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Григорьевич
        </td>
        <td
          style={
            Object {
              "width": "100px",
            }
          }
        >
          -220
        </td>
        <td
          className="blocked"
        >
          block
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    disabled={false}
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа MobileCompany 3`] = `
<div
  className="MobileCompany"
>
  <div
    className="MobileCompanyName"
  >
    Компания: «
    Velcom
    »
  </div>
  <div
    className="MobileControls"
  >
    <input
      className={null}
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      className="activeBtn"
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      className={null}
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
  </div>
  <table>
    <thead>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="MobileClient"
      >
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Иванов
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Иван
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Иванович
        </td>
        <td
          style={
            Object {
              "width": "100px",
            }
          }
        >
          200
        </td>
        <td
          className="active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Сидоров
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Сидор
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Сидорович
        </td>
        <td
          style={
            Object {
              "width": "100px",
            }
          }
        >
          250
        </td>
        <td
          className="active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Петров
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Петр
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Петрович
        </td>
        <td
          style={
            Object {
              "width": "100px",
            }
          }
        >
          180
        </td>
        <td
          className="active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    disabled={false}
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа MobileCompany 4`] = `
<div
  className="MobileCompany"
>
  <div
    className="MobileCompanyName"
  >
    Компания: «
    Velcom
    »
  </div>
  <div
    className="MobileControls"
  >
    <input
      className={null}
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      className={null}
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      className="activeBtn"
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
  </div>
  <table>
    <thead>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="MobileClient"
      >
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Григорьев
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Григорий
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Григорьевич
        </td>
        <td
          style={
            Object {
              "width": "100px",
            }
          }
        >
          -220
        </td>
        <td
          className="blocked"
        >
          block
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    disabled={false}
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа MobileCompany 5`] = `
<div
  className="MobileCompany"
>
  <div
    className="MobileCompanyName"
  >
    Компания: «
    Velcom
    »
  </div>
  <div
    className="MobileControls"
  >
    <input
      className={null}
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      className={null}
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      className="activeBtn"
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
  </div>
  <table>
    <thead>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="MobileClient"
      >
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Григорьев
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Григорий
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Григорьевич
        </td>
        <td
          style={
            Object {
              "width": "100px",
            }
          }
        >
          -220
        </td>
        <td
          className="blocked"
        >
          block
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          <input
            type="text"
          />
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          <input
            type="text"
          />
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          <input
            type="text"
          />
           
        </td>
        <td
          style={
            Object {
              "width": "100px",
            }
          }
        >
          <input
            type="number"
          />
           
        </td>
        <td />
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Сохранить"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Отменить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    disabled={true}
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа MobileCompany 6`] = `
<div
  className="MobileCompany"
>
  <div
    className="MobileCompanyName"
  >
    Компания: «
    Velcom
    »
  </div>
  <div
    className="MobileControls"
  >
    <input
      className={null}
      onClick={[Function]}
      type="button"
      value="Все"
    />
    <input
      className={null}
      onClick={[Function]}
      type="button"
      value="Активные"
    />
    <input
      className="activeBtn"
      onClick={[Function]}
      type="button"
      value="Заблокированные"
    />
  </div>
  <table>
    <thead>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="MobileClient"
      >
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Григорьев
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Григорий
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          Григорьевич
        </td>
        <td
          style={
            Object {
              "width": "100px",
            }
          }
        >
          -220
        </td>
        <td
          className="blocked"
        >
          block
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr
        className="MobileClient"
      >
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          <input
            type="text"
          />
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          <input
            type="text"
          />
        </td>
        <td
          style={
            Object {
              "width": "170px",
            }
          }
        >
          <input
            type="text"
          />
           
        </td>
        <td
          style={
            Object {
              "width": "100px",
            }
          }
        >
          <input
            type="number"
          />
           
        </td>
        <td />
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Сохранить"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="Отменить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    disabled={true}
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;
